app/middleware/authentication.middleware.js

```diff
@@ -18,18 +18,11 @@ exports.authentication = function(request, response, next) {
     userController.getUserByEmail(email)
         .then(function(user) {
             if (!user) {
```

Comment:
I'm thinking maybe we could extract this to `createUserIfFirstLogin` or `createUserOnFirstLogin` or `createUserIfNonexistant` or something along those lines